[yuyang011] iP

yuyang011 opened the pull request:
**Ares**

> “If you spend too much time thinking about a thing, you’ll never get it done.” [Source](https://dansilvestre.com/productivity-quotes/)

Ares frees your mind of having to remember things you need to do. 

- It's,text-based :pencil2:
- easy to learn :open_book:
- ~~FAST~~ _SUPER FAST_ to use :trophy:

All you need to do is,

1. download it from [here](https://github.com/yuyang011/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :smile:

And it is **FREE**!

 Features:

- [X]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

yuyang011 committed:
Renamed to Ares.java

yuyang011 committed:
Removed Duke.java

yuyang011 committed:
Added Echo

yuyang011 committed:
Added Add, List

yuyang011 committed:
Added Task,java, implemented Mark and Unmark

yuyang011 committed:
Added Todos, Events, Deadlines

yuyang011 committed:
Added input.txt and EXPECTED.txt

yuyang011 committed:
Added AresException.java

yuyang011 committed:
Added Delete and updated input testcases

yuyang011 committed:
Add Save feature

yuyang011 committed:
Add Save feature

yuyang011 committed:
Merge branch 'branch-Level-7'

basil-boh commented on src/main/java/Ares.java:
```diff
@@ -0,0 +1,180 @@
+                String input = scanner.nextLine();
+                System.out.println(indent + line);
+                if (input.equals("bye")) {
+                    System.out.println(indent + "Nice chatting with you today\n" + indent + "See you again next time.");
```
Should implement print statements in a separate ui/message class to better adhere to OOP principles of information hiding and tell don't ask.

basil-boh commented on src/main/java/Ares.java:
```diff
@@ -0,0 +1,180 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.*;
```
Should remove java.io.* and specify only packages required

basil-boh commented on src/main/java/Ares.java:
```diff
@@ -0,0 +1,180 @@
+public class Ares {
+    private static final String FILE_PATH = "./data/ares.txt";
+    public static void main(String[] args) {
+        String line = "--------------------------------------------------------------";
```
All fields should be made private

basil-boh commented on src/main/java/Ares.java:
Your main class is handling too much logic, a suggestion would be creating separate classes to handle specific functions like parsing inputs, task management (saving/loading), etc

yuyang011 committed:
Add dates and times

yuyang011 committed:
Merge branch 'branch-Level-8'

quangthangisme commented on src/main/java/AresException.java:
```diff
@@ -0,0 +1,5 @@
+public class AresException extends Exception {
+    public AresException(String message) {
+        super(message);
+    }
+}
```
The `AresException` class seems to be handling too many different types of exceptions. You might consider splitting it into more specific exceptions, such as `UnknownCommandException`, `InvalidArgumentException`, etc.

quangthangisme commented on src/main/java/Ares.java:
```diff
@@ -0,0 +1,203 @@
+                System.out.println(indent + line);
+            }
+        }
+    }
```
You should add a blank line after this line.

quangthangisme commented on src/main/java/Ares.java:
```diff
@@ -0,0 +1,203 @@
+                }
+                String type = parts[0];
+                String description = parts[2];
+                if (type.equals("T")) {
+                    Task newTask = new Todo(description);
+                    if (parts[1].equals("1")) {
+                        newTask.markAsDone();
+                    }
+                    tasks.add(newTask);
+                } else if (type.equals("D")) {
+                    LocalDateTime by = LocalDateTime.parse(parts[3], DATE_TIME_FORMAT);
+                    Task newTask = new Deadline(description, by);
+                    if (parts[1].equals("1")) {
+                        newTask.markAsDone();
+                    }
+                    tasks.add(newTask);
+                } else if (type.equals("E")) {
+                    String[] timings = parts[3].split(" - ");
+                    LocalDateTime from = LocalDateTime.parse(timings[0], DATE_TIME_FORMAT);
+                    LocalDateTime to = LocalDateTime.parse(timings[1], DATE_TIME_FORMAT);
+                    Task newTask = new Event(description, from, to);
+                    if (parts[1].equals("1")) {
+                        newTask.markAsDone();
+                    }
+                    tasks.add(newTask);
+                }
```
There is a lot of repetition in the code when marking the task and adding it to the list. You could refactor the if statements to eliminate redundancy.

quangthangisme commented on src/main/java/Ares.java:
```diff
@@ -0,0 +1,203 @@
+                    break;
+                } else if (input.equals("list")) {
+                    System.out.println(indent + "Here are the tasks in your list:");
+                    for (int i = 0; i < tasks.size(); i++){
```
You should add a whitespace before `{`.

quangthangisme commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,27 @@
+
+public class Event extends Task {
+
+    protected LocalDateTime from;
```
The attribute name could be more precise as a noun phrase. You might consider renaming it to `startTime`.

quangthangisme commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,27 @@
+    @Override
+    public String toString() {
+        DateTimeFormatter outputFormat = DateTimeFormatter.ofPattern("MMM dd yyyy HHmm");
+        return "[E]" + super.toString() + " (from: " + this.from.format(outputFormat) + " to: " + this.to.format(outputFormat) + ")";
```
This line is exceeding the character limit of 100 characters. You can break it into multiple lines like this.
```suggestion
        return "[E]" + super.toString() + " (from: " + this.from.format(outputFormat) 
                + " to: " + this.to.format(outputFormat) + ")";
```

quangthangisme reviewed (commented):
Overall, I found your code a bit hard to read as the `Main` class assumes too many responsibilities. You might consider breaking it into components such as `Storage`, `Ui`, and `TaskList`. Also, there are several coding standard violations you may want to address.

yuyang011 committed:
Refactor A-MoreOOP

yuyang011 committed:
Merge branch 'Branch-A-MoreOOP'

yuyang011 committed:
Merge branch 'add-gradle-support'

yuyang011 committed:
Merge branch 'add-gradle-support

yuyang011 committed:
Change folder directory

yuyang011 committed:
Add A-JUnit

yuyang011 committed:
Merge branch 'branch-A-JUnit'

yuyang011 committed:
Create JAR file

yuyang011 committed:
Add Find feature

yuyang011 committed:
Merge branch 'branch-Level-9'

yuyang011 committed:
Edit code to follow coding standard

yuyang011 committed:
Merge branch 'branch-A-CodingStandard'

yuyang011 committed:
Add javadoc comments

yuyang011 committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/ares/task/Task.java
#	src/main/java/ares/task/TaskList.java
#	src/main/java/ares/ui/Ui.java

yuyang011 committed:
Add Checkstyle

yuyang011 committed:
Merge branch 'A-CheckStyle'

yuyang011 committed:
Add Gui

yuyang011 committed:
Merge branch 'branch-Level-10'

yuyang011 committed:
Add assertions

Assertations are added to the commands to ensure that parameters
are not null, task index is not negative and file directory is
created.

Adding assertions into the program helps verify that the program is
running correctly.

yuyang011 committed:
Merge pull request #2 from yuyang011/branch-A-Assertions

Add assertions

yuyang011 committed:
Refactor command codes

Ui class have some code duplication, thus extracting the duplicate
codes and consolidating them will help improve code quality.

Mark and Unmark Command have some long methods, extracting them
allows us to be able to read the code more easily.

yuyang011 committed:
Merge pull request #3 from yuyang011/branch-A-CodeQuality

Refactor command codes

yuyang011 committed:
Add B-ViewSchedules

yuyang011 committed:
Merge branch 'branch-BCD-Extension'

yuyang011 committed:
Edit codes to ensure it follows coding convention

yuyang011 committed:
Add Readme and Ui.png

yuyang011 committed:
Update Readme

yuyang011 committed:
Update README.md

yuyang011 committed:
Update README.md

yuyang011 committed:
Update Readme

yuyang011 committed:
Update README.md

yuyang011 committed:
Update README.md

yuyang011 committed:
Update Readme

yuyang011 committed:
Update View Command
